cython/gtsam/utils/logging_optimizer.py

```diff
@@ -46,5 +46,6 @@ def gtsam_optimize(optimizer,
     def check_convergence(optimizer, current_error, new_error):
         return (optimizer.iterations() >= params.getMaxIterations()) or (
             gtsam.checkConvergence(params.getRelativeErrorTol(), params.getAbsoluteErrorTol(), params.getErrorTol(),
-                                   current_error, new_error))
+                                   current_error, new_error)) or (optimizer.lambda_() > params.getlambdaUpperBound())
```

Comment:
Should be `optimizer.lambda()` not `lambda_()` (no underscore).

Reply:
should have underscore as lambda is a reserved word